app/src/main/java/org/oppia/android/app/onboarding/AudioLanguageFragmentPresenter.kt

```diff
@@ -52,22 +87,78 @@ class AudioLanguageFragmentPresenter @Inject constructor(
       activity.finish()
     }
 
-    val adapter = ArrayAdapter(
-      fragment.requireContext(),
-      R.layout.onboarding_language_dropdown_item,
-      R.id.onboarding_language_text_view,
-      audioLanguageSelectionViewModel.availableAudioLanguages
+    audioLanguageSelectionViewModel.availableAudioLanguages.observe(
+      fragment,
+      { languages ->
+        val adapter = ArrayAdapter(
+          fragment.requireContext(),
+          R.layout.onboarding_language_dropdown_item,
+          R.id.onboarding_language_text_view,
+          languages
+        )
+        binding.audioLanguageDropdownList.setAdapter(adapter)
+      }
     )
 
     binding.audioLanguageDropdownList.apply {
-      setAdapter(adapter)
-      setText(
-        audioLanguageSelectionViewModel.defaultLanguageSelection,
-        false
-      )
       setRawInputType(EditorInfo.TYPE_NULL)
+
+      onItemClickListener =
+        AdapterView.OnItemClickListener { _, _, position, _ ->
+          adapter.getItem(position).let { selectedItem ->
+            if (selectedItem != null) {
+              selectedLanguage = selectedItem as String
+            }
+          }
+        }
+    }
+
+    binding.onboardingNavigationContinue.setOnClickListener {
+      updateSelectedAudioLanguage(selectedLanguage, profileId)
     }
 
     return binding.root
   }
+
+  private fun observePreselectedLanguage() {
+    audioLanguageSelectionViewModel.languagePreselectionLiveData.observe(
+      fragment,
+      { selectedLanguage -> setSelectedLanguage(selectedLanguage) }
+    )
+  }
+
+  private fun setSelectedLanguage(selectedLanguage: String) {
+    this.selectedLanguage = selectedLanguage
+    audioLanguageSelectionViewModel.selectedAudioLanguage.set(selectedLanguage)
+  }
+
+  private fun updateSelectedAudioLanguage(selectedLanguage: String, profileId: ProfileId) {
+    val audioLanguage =
+      appLanguageResourceHandler.getAudioLanguageFromLocalizedName(selectedLanguage)
+    profileManagementController.updateAudioLanguage(profileId, audioLanguage).toLiveData()
+      .observe(fragment) {
+        when (it) {
+          is AsyncResult.Success -> {
+            val intent = HomeActivity.createHomeActivity(fragment.requireContext(), profileId)
+            fragment.startActivity(intent)
+            fragment.activity?.finishAffinity()
```

Comment:
Why use finishAffinity here? Seems worthwhile to add a comment, and maybe a test (to ensure its being used over just finish()).